File: frontend/src/types/payment.ts
import type { User } from './user';

export type PaymentStatus = 'unpaid' | 'paid' | 'overdue';
export type FeeType = 'property' | 'parking' | 'utilities';

export interface Payment {
  id: number;
  userId: number;
  feeType: FeeType;
  amount: number;
  month: string;
  status: PaymentStatus;
  dueDate?: string;
  paidAt?: string;
  createdAt: string;
  overdue?: boolean;
  user?: User;
}

export interface PaymentOverdueStats {
  overdueAmount: number;
  overdueCount: number;
}

export interface PaymentRemindResult {
  successCount: number;
  failCount: number;
  totalCount: number;
}

File: backend/src/main/java/com/smartestate/entity/Payment.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("payments")
public class Payment {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String feeType;
    private BigDecimal amount;
    private String month;
    private String status;
    private LocalDateTime dueDate;
    private LocalDateTime paidAt;
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private User user;

    @TableField(exist = false)
    private Boolean overdue;

    public boolean isOverdue() {
        if ("paid".equals(status)) {
            return false;
        }
        if ("overdue".equals(status)) {
            return true;
        }
        if (dueDate != null) {
            return LocalDateTime.now().isAfter(dueDate);
        }
        return false;
    }
}
